### ISSUE TITLE
mAP is error??

### ISSUE THREAD

**Issue by sbbug** [ultralytics/yolov5#2448]
## ❔Question
Why is best.pt not the highest map when tested by test.py.

**Comment by glenn-jocher** [comment]
@sbbug best.pt is saved at every new highest fitness epoch:
https://github.com/ultralytics/yolov5/blob/886f1c03d839575afecb059accf74296fad395b6/train.py#L393-L394

fitness is defined by a weighted combination of metrics in metrics.py

**Comment by sbbug** [comment]
@glenn-jocher Thank you for your prompt reply!
Yean，but, why is it that when I use the best.pt test it doesn't look like the mAP I trained to see, which is wrong

**Comment by glenn-jocher** [comment]
@sbbug if you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

**Comment by sbbug** [comment]
The highest MAP you see during training is:
`     39/49     4.83G   0.03183   0.04027   0.01029   0.08239        77       672
bn_c 97
               Class      Images     Targets           P           R      mAP@.5
                 all         537    6.08e+03        0.25       0.532       0.395       0.173`

But when I test it in the test.py file with the saved BEST.PT, I get the following mAP:
Namespace(augment=False, batch_size=8, conf_thres=0.001, data='data/uva.yaml', device='0', exist_ok=False, img_size=672, iou_thres=0.6, modal_name='visible', name='exp', project='runs/test', save_conf=False, save_hybrid=False, save_json=False, save_txt=True, single_cls=False, task='val', verbose=False, weights='/home/shw/code/yolov5-master/runs/train/exp674/weights/best.pt')
Using torch 1.7.0 CUDA:0 (GeForce RTX 2080 Ti, 11019MB)

Fusing layers... 
Model Summary: 232 layers, 7265397 parameters, 0 gradients, 16.9 GFLOPS
Scanning '/home/shw/data/UVA/val/annotations/07_07-39-00-07-42-00-shw.cache' forlen(dataset): 537

Scanning '/home/shw/data/UVA/val/annotations/07_07-39-00-07-42-00-shw.cache' for

               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
               Class      Images     Targets           P           R      mAP@.5
                 all         537    6.08e+03       0.249       0.532        0.38       0.167
Results saved to runs/test/exp324

Process finished with exit code 0
why?

**Comment by sbbug** [comment]
@glenn-jocher 
0.395 and 0.38, this is confused!

**Comment by glenn-jocher** [comment]
@sbbug testing during training runs at --iou 0.60. test.py directly called uses the argparser default of 0.65. You might try varying --iou to see if they match.
